client/src/components/events-components/EventList.tsx

```diff
@@ -1,13 +1,37 @@
+import { useEffect, useState } from "react";
 import EventCard from "./EventCard";
+import { getAllEvents } from "@/utils/eventService";
+
+interface Event {
+  created_at: string;
+  description: string;
+  eventImage: string;
+  eventTime: string;
+  location: string;
+  latitude: number;
+  longitude: number;
+  organizerId: number;
+  id: number;
+  sport: string;
+  updated_at: string;
+}
 
 const EventList = () => {
-  const events = [{ name: "john" }, { name: "Hernyr" }];
+  const [events, setEvents] = useState([]);
+
+  useEffect(() => {
+    const fetchEvents = async () => {
+      const retrievedEvents = await getAllEvents();
+      setEvents(retrievedEvents);
+    };
+    fetchEvents();
+  }, []);
   return (
-    <section className="flex justify-start">
-      {events.map((event) => {
-        return <EventCard event={event} />;
+    <ul className="flex flex-col sm:flex-row justify-center sm:justify-start flex-wrap items-center">
+      {events.map((event: Event) => {
+        return <EventCard key={event.id} event={event} />;
```

Comment:
```suggestion
        return <li><EventCard key={event.id} event={event} /><li>;
```

use `ul` and `li` together